core/app/models/spree/promotion.rb

```diff
@@ -112,13 +112,11 @@ def activate(order:, line_item: nil, user: nil, path: nil, promotion_code: nil)
       # If an action has been taken, report back to whatever activated this promotion.
       action_taken = results.include?(true)
 
-      if action_taken
-        # connect to the order
-        order_promotions.find_or_create_by!(
-          order_id: order.id,
-          promotion_code_id: promotion_code.try!(:id)
-        )
-      end
+      # connect to the order
+      order_promotions.find_or_create_by!(
+        order_id: order.id,
+        promotion_code_id: promotion_code.try!(:id)
+      )
 
       action_taken
```

Comment:
Not sure what the return value should be after this change. Previously I had thought of it like AR's `save`, `true` means the action was successful, `false` means nothing happened.

`true` now means the promotion was attached and is currently active.
`false` now means the promotion was attached, but is inactive for the time being.

Maybe this is fine, but I thought I'd bring it up.